File: pothole-indicator.py
def Pothole_indicator(R):
    sub_arr =  []
    pothole_indicator = 0

    for num in R:
        if num != 0:
            sub_arr.append(num)
        else:
            if sub_arr:
                print(sub_arr)
                RK = max(sub_arr)*len(sub_arr)
                if RK > pothole_indicator:
                    pothole_indicator = RK
                sub_arr = []
    if sub_arr:
        print(sub_arr)
        RK = max(sub_arr)*len(sub_arr)
        if RK > pothole_indicator:
            pothole_indicator = RK
        sub_arr = []

    print(f'Pothole Indicator: {pothole_indicator}')
Pothole_indicator([1, 4, 1, 0, 5, 2, 3, 0, 8])

File: PotholeIndicator.js
function PotholeIndicator(R){
    let sub_arr = []
    let portholeIndicator = 0

    function calculate(){
        let RK = ((Math.max.apply(null, sub_arr)) * (sub_arr.length))
        if (RK > portholeIndicator){
            portholeIndicator = RK
        }
        sub_arr = []
    }
    for (let num of R){
        if (num !== 0){
            sub_arr.push(num)
        }
        else{
            console.log(sub_arr)
            calculate()
        }
    }
    if (sub_arr.length > 1){
        console.log(sub_arr)
        calculate()
    }
    console.log(portholeIndicator)
}
PotholeIndicator([9, 8, 7, 0, 0, 0, 2, 3, 6, 4])